### PR TITLE
Feature - Shipping Addon

### LINKED ISSUE #1391: Update table rate shipping package to add Filament screens

### PR DESCRIPTION
**Pull request by alecritson**

### CODE REVIEW

**Comment by ryanmitchell** [comment]
Oh... so the package is being brought into core?

**Comment by glennjacobs** [comment]
> Oh... so the package is being brought into core?

Yeah, for 1.x

**Comment by alecritson** [comment]
> Oh... so the package is being brought into core?

It's just being added as part of the mono repo, it'll still function as an add-on. So that it's easier to keep track of changes and provide any fixes in one hit rather than being spread out and having to think what might need fixing across repos

**Comment by ryanmitchell** [comment]
Nice

**Comment by glennjacobs** [comment]
image


Also, the price field is very wide...

**Comment by glennjacobs** [comment]
Does the Base Price need to consider currency also?

**Comment by glennjacobs** [comment]
image

Can the name and description be translated?

**Comment by alecritson** [comment]
> image
> 
> Also, the price field is very wide...

## Base price

Can be changed to just "Price" like it is on product pricing?

## Min Spend.

Would this be something that admins can choose? Currently, it's on the sub-total since the total will account for shipping costs already applied and could put itself over the threshold.

## Currency

Yeah, it should however, nothing in the panel currently handles currencies with pricing and it didn't seem like it should be part of this PR. It would need to be updated when this is done properly.

## Pricing

ShippingRates are purchasables so they use the same logic as other purchasables.

## Can name/desc be translated?

Are you asking if they are capable of being translated or if they can be made so they can be translated? If the former, then no, just a text column.

**Comment by glennjacobs** [comment]
> ## Base price
> Can be changed to just "Price" like it is on product pricing?

OK.


> ## Min Spend.
> Would this be something that admins can choose? Currently, it's on the sub-total since the total will account for shipping costs already applied and could put itself over the threshold.

As you say, it will have to run off the sub-total. The confusing bit is whether it considers tax or not, as some will be storing prices inclusive of tax. Maybe we have helper text that specifies what it is, based upon whether `lunar.pricing.stored_inclusive_of_tax` is set to true or not. And of course, the actual shipping logic will need to be updated also.

> ## Currency
> Yeah, it should however, nothing in the panel currently handles currencies with pricing and it didn't seem like it should be part of this PR. It would need to be updated when this is done properly.

Right, so we need a proper pricing field/component to handle this. I'm happy to come back to this, as long as we make an issue to remember :-)

> ## Pricing
> ShippingRates are purchasables so they use the same logic as other purchasables.

Again, I think we might just want to add some helper text to reassure the user.

> ## Can name/desc be translated?
> Are you asking if they are capable of being translated or if they can be made so they can be translated? If the former, then no, just a text column.

I think we will need to add this, but I think we can come back to it as long as we make an issue to remind us.

File: packages/table-rate-shipping/lang/en/shipping-methods.php [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+return [
+    '',
```
Comment (glennjacobs) [inline packages/table-rate-shipping/lang/en/shipping-methods.php]: What is this file for? It's empty and shouldn't language files reside in the resources folder?

**Pull request merged**